### DISCUSSION TITLE
Installation script for common Linux Variants

### DISCUSSION THREAD

**Discussion by kornyhiv** [blakeblackshear/frigate#9208] [Show and tell]
Good Morning Everyone,

I have now been using Frigate for some time, and I love this project. Thank you to all who have invested, and provided continued development, and support for this platform.

With that said, the installation of the software maybe difficult for novice people not familiar with linux, docker, and other variables. In an effort to make things easier, and enhance the reach of Frigate - I have created a bash script to easily install Frigate on Ubuntu Linux.

This can be used to install Frigate or as a template to either further enhance or create a new script all together.

My intention is only to further propagate the reach and utilization of this software to hopefully add additional support and development time with the increased user base for non techies (Novice people).

Instructions on a brand new Ubuntu based machine. (Minimal Install works great).

1. Install Ubuntu Server, minimal footprint.
2. Select installation of SSH Server during the Ubuntu installation Wizard.
3. After installation completes, use Putty to connect to the new Ubuntu installation. 
4. Update Ubuntu. Sudo apt-get update and Sudo apt-get upgrade
5. Install nano (text editor) - Sudo apt-get install nano
6. Create frigate.sh file. nano frigate.sh
7. Copy and paste the bash script here and save changes into frigate.sh
8. Change permission on file/execution rights. chmod +x frigate.sh
9. Start Frigate installation. sudo ./frigate.sh start
10. Dependencies will install, and you may need run ./frigate.sh start a few times until completion.

Notes:

Start Command - Installs Frigate or runs frigate if already installed before using previous parameters selected (frigate_installation_settings file)

Stop Command - Stops running Frigate docker container. Start command can be used to start it again.

Delete Command - Stops running Frigate container, deletes the Frigate container, and deletes the frigate_installation_settings file.
(Delete command does not delete config folder/file, or Media folder.)

Frigate_Installation_Settings file - this keeps the variables used during the installation wizard, so that if the container is stopped, service is stopped, etc. The start command can be used to start the Frigate container again with the existing variables that were input on the wizard during installation, without asking for those again.

Basic installation which can be run as a quick test to verify config using CPU. Is "no" to USB Coral prompt, and "no" for NVIDIA GPU for video decoding.

The Config.yml file included is a template so you don't get the pesky no configuration file found that many novice people get when trying to install Frigate for the first time, or troubleshooting why it doesn't start.

The Config Tab inside of Frigate (GUI) once is running is recommended, as it contains YAML structure verification before submitting a change or making a mistake. You can then quickly copy and paste your config file in most cases to get up and running.

Your config file could be pasted on the script between the "mqtt", and "FPS: 7" if needed, however it is highly recommended top use the Frigate GUI Config tab once it is running.

The Wizard also prompts for NVIDIA GPU usage, and USB Coral.

All this does, is it adds docker run commands depending on the user submission.

USB Coral:
       --device /dev/bus/usb:/dev/bus/usb"

NVIDIA GPU for video decoding:
        --gpus all \
        -e NVIDIA_VISIBLE_DEVICES=all \
        -e **NVIDIA_DRIVER_CAPABILITIES=compute,video,utility"**

However, these are in because they match my systems. These and any other variable can be easy edited or removed to better match your setups. Keep this in mind, because if your system differs on the Hardware Acceleration or other variables these may need to be changed for docker to fully utilize your hardware.

The script will  also automatically retrieve the latest beta version if selected. The script after installation will also provide local IPv4 address for novice users if not using localhost.

An effort has been made to comment all lines to clearly explain the action or desired function for the code. This will make it easier to modify and match your needs.

Again, just trying to provide one more tool or the basis of a new idea to create something similar.

**Please use at your own risk,**

Here are a few screenshots:
Start Command:
![image](https://github.com/blakeblackshear/frigate/assets/29285614/cd9d47e5-57f7-4e2f-95de-e6a4865d208e)

Stop Command:
![image](https://github.com/blakeblackshear/frigate/assets/29285614/920c6eea-4067-4469-9229-37dcdfb53f12)

Delete Command:
![image](https://github.com/blakeblackshear/frigate/assets/29285614/a4b33512-189e-4532-ba4f-f347b434db96)

Edit: 08/11/2025

Script:

https://github.com/kornyhiv/frigatenvr-easyinstallation

Several improvements including NVIDIA packages and drivers.

**Comment by NickM-27** [comment]
Moving this to discussion as this is not a feature request

**Comment by kd4e** [comment]
_Step 6: Create frigate.sh file. nano frigate.sh_

Should these be created in /usr/local/bin or /opt/bin, please?

Note 1: I'm trying this in MX Linux 23.2 (a Debian derivative).
Note 2: I'm on a Dell OptiPlex 7060 CPU: Intel 8500, 64G ram, 256G ssd,  and a 2TB nvme
Note 3: I plan to host about 12 cameras, as well as some additional security-related devices.
Note 4: I also plan to install openHAB.

Thanks!

**Comment by jasonrarnold** [comment]
This is Awesome!!! 
Worked for me!
Thank you!

**Comment by kornyhiv** [comment]
Here is my current updated script.

If you do not have an NVIDIA Card select no on the prompt, and select yes for USB coral if you have one. Also included is the Frigate Plus API parameter - Purchase to support developer and improve the standard coral model from Google.

You may need to make additional modifications to match your environment.

https://pastebin.com/pCaUG5r6

**Comment by kornyhiv** [comment]
https://github.com/kornyhiv/frigatenvr-easyinstallation

Several improvements including NVIDIA packages and drivers.
